lib/spack/env/cc

```diff
@@ -98,25 +98,25 @@ case "$command" in
     cpp)
         mode=cpp
         ;;
-    cc|c89|c99|gcc|clang|icc|pgcc|xlc)
+    cc|c89|c99|gcc|clang|icc|pgcc|xlc|xlc_r)
```

Comment:
The _r versions set the appropriate preprocessor defines for multi-threaded compilation, and supply the appropriate threaded library names to the linker.

So, if we remove the option for 'xlc_r', spack has to provide additional option for those who need xl_r.  We are also learning about the spack, so please let us know what would be the best way.